extras/convenience/MapkitExtensions.swift

```diff
@@ -58,18 +57,46 @@ Example:
 
 */
 
+public let PeliasIDKey: String = "PeliasOSMIDKey"
+public let PeliasDataSourceKey: String = "PeliasDataSourceKey"
+
+class PeliasMapkitAnnotation: NSObject, MKAnnotation {
+  
+  let coordinate: CLLocationCoordinate2D
+  let title: String?
+  let subtitle: String?
+  let data: [String: AnyObject]?
+  
+  init(coordinate: CLLocationCoordinate2D, title: String?, subtitle: String?, data: [String:AnyObject]?) {
+    self.coordinate = coordinate
+    self.title = title
+    self.subtitle = subtitle
+    self.data = data
+  }
+}
+
+extension PeliasPlaceQueryItem {
+  init?(annotation: PeliasMapkitAnnotation, layer: LayerFilter) {
+    guard let place = annotation.data?[PeliasIDKey] as? String else { return nil }
+    guard let source = SearchSource(rawValue: annotation.data?[PeliasDataSourceKey] as? String ?? "") else { return nil }
+    self.placeId = place
+    self.dataSource = source
+    self.layer = layer
+  }
+}
+
 extension PeliasResponse {
-  func parsedMapItems() -> [MKMapItem]? {
+  func parsedMapItems() -> [PeliasMapkitAnnotation]? {
     //TODO: This should get refactored into eventually being a real GeoJSON decoder, and split out the MapItem creation
-    var mapItems = [MKMapItem]()
+    var mapItems = [PeliasMapkitAnnotation]()
     if let jsonDictionary = parsedResponse?.parsedResponse {
       let featuresArray = jsonDictionary["features"] as! [[String:AnyObject]]
       for feature in featuresArray {
         //Address Dictionary for Placemark Creation
         let featureProperties = feature["properties"] as! [String:AnyObject]
         var addressDictionary = [String:String]()
-        addressDictionary[CNPostalAddressCountryKey] = featureProperties["country"] as? String
-        
+        addressDictionary[PeliasIDKey] = featureProperties["id"] as? String
```

Comment:
I do for the "parsed" side of things, as you can see. This particular area needs improvement in general, because basically I'm having to traverse a GeoJSON schema. So the upside here is since Pelias follows the schema, these names won't change much if at all. 

However, what I really wanna do is abstract this out and make use of a real GeoJSON parser, like this one https://github.com/mattt/GeoJSONSerialization

I'm open to thoughts over here on the topic: https://github.com/pelias/pelias-ios-sdk/issues/4